### PR TITLE
Fix error on membership imports

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix error on membership imports

Possibly related to https://civicrm.stackexchange.com/questions/42502/membership-import-fails-on-queue-runner-after-upgrade-to-5-52

Before
----------------------------------------
Importing the start date is hitting an error
image

After
----------------------------------------
Error should be fixed

Technical Details
----------------------------------------
I'd like to do a better master fix 

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Member/Import/Parser/Membership.php [hunk 1]
```diff
@@ -461,7 +461,7 @@ protected function setFieldMetadata(): void {
    * @return string[]
    */
   protected function getOddlyMappedMetadataFields(): array {
-    $uniqueNames = ['membership_id', 'membership_contact_id'];
+    $uniqueNames = ['membership_id', 'membership_contact_id', 'membership_start_date', 'membership_join_date', 'membership_end_date', 'membership_source', 'member_is_override', 'member_is_test', 'member_is_pay_later', 'member_campaign_id'];
```
Comment (seamuslee001) [inline CRM/Member/Import/Parser/Membership.php:464]: why member_is_test member_x not membership?

Comment (eileenmcnaughton) [reply CRM/Member/Import/Parser/Membership.php:464]: I took it from the `Membership.xml`

**Comment by seamuslee001** [comment]
ok seems right to me then

**Comment by eileenmcnaughton** [comment]
Port [24358](https://github.com/civicrm/civicrm-core/pull/24358)

**Pull request merged**